File: ApiSuvesaPos/Negocio/Logica/TokenPermisoNegativo.vb
Namespace Logica
    Public Class TokenPermisoNegativo

        Private db As Datos.Class.TokenPermisoNegativo

        Sub New()
            Me.db = New Datos.Class.TokenPermisoNegativo
        End Sub

        Public Function ConsultarToken() As List(Of Datos.Models.ViewTokenSolicitado)
            Return Me.db.ListaTokenSolicitadosdelDia
        End Function


        Public Function SolicitarToken(idusuario As String, codigo As Decimal) As String
            Return Me.db.SolicitarTokenVentaNegativa(idusuario, codigo)
        End Function

        Public Function ValidarToken(idusuario As String, codigo As Decimal, token As String) As List(Of Datos.Models.ViewTokenSolicitado)
            Return Me.db.ConsultarToken(idusuario, codigo, token)
        End Function



    End Class
End Namespace

File: ApiSuvesaPos/Datos/Class/TokenPermisoNegativo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Datos.Models;
using Microsoft.EntityFrameworkCore;

namespace Datos.Class
{
	public class TokenPermisoNegativo
	{
		private SeePOSContext entities;

		public TokenPermisoNegativo()
		{
			entities = new SeePOSContext();
		}

		public int SolicitarTokenVentaNegativa(string IdUsuario, decimal Codigo) // registro de TemperaturaCamara
		{
			try
			{
				Random rand = new Random();
				var Token = new Datos.Models.TokenPermisoNegativo();
				Token.IdUsuarioSolicita = IdUsuario;
				Token.Codigo = Codigo;
				Token.Fecha = DateTime.Now;
				Token.Token = rand.Next(0, 999).ToString();

				entities.TokenPermisoNegativos.Add(Token);
				return entities.SaveChanges();
			}
			catch (Exception ex)
			{
				throw ex;
			}
		}

		public List<ViewTokenSolicitado> ConsultarToken(string IdUsuario, decimal Codigo, string Token)
		{
			try
			{
				List<ViewTokenSolicitado> result;
				var temp = from c in entities.ViewTokenSolicitados
						   where c.IdUsuarioSolicita == IdUsuario && c.Codigo == Codigo && c.Token == Token
						   select c;
				result = temp.ToList<ViewTokenSolicitado>();

				if (result.Count > 0)
				{
					return result;
				}
				else
				{
					return result = null;
				}

			}
			catch (Exception ex)
			{
				throw ex;
			}
		}

		public List<ViewTokenSolicitado> ListaTokenSolicitadosdelDia()
		{
			try
			{			
				List<ViewTokenSolicitado> result;
				var temp = from c in entities.ViewTokenSolicitados
						   where c.Fecha.Day == DateTime.Now.Day && c.Fecha.Month == DateTime.Now.Month && c.Fecha.Year == DateTime.Now.Year
						   select c;
				result = temp.ToList<ViewTokenSolicitado>();

				if (result.Count > 0)
				{
					return result;
				}
				else
				{
					return result = null;
				}

			}
			catch (Exception ex)
			{
				throw ex;
			}
		}


	}
}


